core/src/main/java/de/danoeh/antennapod/core/service/GpodnetSyncService.java

```diff
@@ -78,24 +86,20 @@ public int onStartCommand(Intent intent, int flags, int startId) {
             }
             if(syncSubscriptions || syncActions) {
                 Log.d(TAG, String.format("Waiting %d milliseconds before uploading changes", WAIT_INTERVAL));
-                syncWaiterThread.restart();
+                int syncActionId = syncActionCount.incrementAndGet();
```

Comment:
you are chaning a lot of stuff here without a test safety net.
are you sure that this simplification does not introduce new issues or weird behavior in edge cases?

Reply:
apart from that, the code is still pretty ugly...
does this even need to be changed for api 26? it seems to me that this should rather be in a separate PR

Reply:
This needs changes because the old way (canceling threads) could not return the status of every scheduled intent.